### PR TITLE
Fixes wrong attendance order

### PR DESCRIPTION
**Pull request by ghatighorias**
ordering was forgotten to be added for the teacher and was just added for the coordinator. this commit ensures the ordering based on the student name and family_name to be added for the teacher as well

root cause :
in the show we use "get_teacher_course_attendances" while in fill we use "get_course_attendances" function to retrive all the attendances. both have to have the same sorting order so the rendered order of the students and attendance be the same

The same problem won't happen for student cause he only sees his own attendance and need no sorting on name or family_name

### CODE REVIEW

File: lib/course_planner/attendances/attendances.ex [hunk 1]
```diff
@@ -27,10 +27,12 @@ defmodule CoursePlanner.Attendances do
       join: s in assoc(oc, :students),
       join: c in assoc(oc, :classes),
       join: a in assoc(c,  :attendances),
+      join: as in assoc(a, :student),
```
Comment (ggpasqualino) [inline lib/course_planner/attendances/attendances.ex]: I don't get why these two new joins are necessary

Comment (ghatighorias) [reply lib/course_planner/attendances/attendances.ex]: we have had this problem with "get_course_attendances" and fixed it with this joins and sort them
in the show and fill I use offered_course.students to create a part of the table but then iterate throw offered_course.class.attendance ( which has student link too ) to check the attendance of each student, so both should be sorted the same in order columns and attendance data to match

**Comment by ghatighorias** [comment]
even this solution is not complete, adding some more student makes it completely messy

**Comment by ghatighorias** [comment]
The last comment fixes all the issues, the ordering should have been done from the beginning based on attendance student and not offered_course student and likewise drawing the tables in the show and fill pages

**Pull request merged**